### PR TITLE
CORE-19093: Ensure propagation of Tracing Data from E2E test call

### PR DESCRIPTION
**Pull request by vkolomeyko**
After the change on E2E test side:

```
16:36:44.453 [ForkJoinPool-1-worker-1] INFO  ClusterBuilder - B - Sending flowStart, vNode: 'A80B3DD2BBB9', clientRequestId: 'start_REST_flow-0', traceId: '65aaa51ce9a9d64190760b1fae7391dc'
```

On the cluster side:

```
2024-01-19 16:36:44.469 [qtp1411078979-2168] INFO  net.corda.rest.server.impl.context.ContextUtils.POST {http.method=POST, http.path=/api/v5_2/flow/A80B3DD2BBB9, http.user=admin, request_id=start_REST_flow-0, spanId=f89e8b77a1fd20d5, traceId=65aaa51ce9a9d64190760b1fae7391dc} - Servicing POST request to '/api/v5_2/flow/A80B3DD2BBB9' and invoking  method "startFlow"
2024-01-19 16:36:44.629 [qtp1411078979-2168] INFO  net.corda.rest.server.impl.context.ContextUtils.GET {http.method=GET, http.path=/api/v5_2/flow/A80B3DD2BBB9/start_REST_flow-0/result, http.user=admin, spanId=363a56e06d18d560, traceId=65aaa51c9daa9eac363a56e06d18d560} - Servicing GET request to '/api/v5_2/flow/A80B3DD2BBB9/start_REST_flow-0/result' and invoking  method "getFlowResult"
2024-01-19 16:36:44.630 [qtp1411078979-2168] INFO  net.corda.rest.server.impl.context.ContextUtils.GET {http.method=GET, http.path=/api/v5_2/flow/A80B3DD2BBB9/start_REST_flow-0/result, http.user=admin, spanId=363a56e06d18d560, traceId=65aaa51c9daa9eac363a56e06d18d560} - Error invoking path '/api/v5_2/flow/{holdingidentityshorthash}/{clientrequestid}/result' - Failed to find the flow status for holdingId = A80B3DD2BBB9 and clientRequestId = start_REST_flow-0.
2024-01-19 16:36:45.097 [flow-event-mediator-thread-0] INFO  net.corda.flow.pipeline.handlers.events.StartFlowEventHandler {corda.client.id=start_REST_flow-0, flow.id=1238ac04-e1be-4f37-b9f2-fbf1298b8834, request_id=start_REST_flow-0, spanId=256f77655fbcca17, traceId=65aaa51ce9a9d64190760b1fae7391dc, vnode.id=A80B3DD2BBB9} - Flow [1238ac04-e1be-4f37-b9f2-fbf1298b8834] started
```

I.e. `traceId` is correctly retained and can be matched to the actual E2E test executed.

This PR also includes few renames and simplifications. There should be no changes to the external API we supply to the E2E tests driver.

### CODE REVIEW

File: testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/ClusterBuilder.kt [hunk 1]
```diff
@@ -664,10 +683,16 @@ class ClusterBuilder {
         flowClassName: String,
         requestData: String
     ): SimpleResponse {
-        return post(
-            "/api/$REST_API_VERSION_PATH/flow/$holdingIdentityShortHash",
-            flowStartBody(clientRequestId, flowClassName, requestData)
-        )
+        return trace("flowStart") {
```
Comment (vkolomeyko) [inline testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/ClusterBuilder.kt:686]: The "meat" of the change.

File: testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/UnirestHttpsClient.kt [hunk 2]
```diff
@@ -39,15 +41,27 @@ class UnirestHttpsClient(private val endpoint: URI, private val username: String
 
     override fun post(cmd: String, body: String): SimpleResponse {
         val url = endpoint.resolve(cmd).toURL().toString()
-
         val response = Unirest.post(url)
             .basicAuth(username, password)
             .body(body)
+            .headers(obtainTracingHeaders())
```
Comment (vkolomeyko) [inline testing/e2e-test-utilities/src/main/kotlin/net/corda/e2etest/utilities/UnirestHttpsClient.kt:47]: Second important point: Include tracing information into HTTP headers when sent from client side.

**Comment by vkolomeyko** [comment]
build e2e

**Comment by filipesoliveira** [comment]
It looks good!

Example from https://ci02.dev.r3.com/blue/organizations/jenkins/Corda5%2Fcorda-runtime-os-combined-worker-e2e-tests/detail/PR-5472/3/tests/:
Get the traceId from the test:
image

Get all logs related to the test:
image

I can now see the flow starting and completing and all the logs in between without much effort.

**Review by filipesoliveira — APPROVED** [review]

**Review by mag1c-48 — APPROVED** [review]
LGTM - build files only

**Pull request merged**